From 8a479f9dd82ba86a7122661fb10e5f3cf8ffa747 Mon Sep 17 00:00:00 2001
From: Maarten Bent <MaartenBent@users.noreply.github.com>
Date: Thu, 16 Aug 2018 20:08:56 +0200
Subject: Fix c++11-narrowing error when using clang on Windows

Case value 0xfffffd9f results in the following error:
error: case value evaluates to 4294966687, which cannot be narrowed to type
'DISPID' (aka 'long') [-Wc++11-narrowing]

(cherry picked from commit fd81223a2f878757158b5266c362da71d43abee5)
---
 src/msw/mediactrl_am.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/msw/mediactrl_am.cpp b/src/msw/mediactrl_am.cpp
index f5b9ed4a61..674da36e46 100644
--- a/src/msw/mediactrl_am.cpp
+++ b/src/msw/mediactrl_am.cpp
@@ -2179,7 +2179,9 @@ void wxAMMediaBackend::Move(int WXUNUSED(x), int WXUNUSED(y),
 //---------------------------------------------------------------------------
 void wxAMMediaEvtHandler::OnActiveX(wxActiveXEvent& event)
 {
-    switch(event.GetDispatchId())
+    // cast to unsigned long to fix narrowing error with case 0xfffffd9f
+    // when using clang
+    switch (static_cast<unsigned long>(event.GetDispatchId()))
     {
 #ifndef __WXWINCE__
     case 0x00000001: // statechange in IActiveMovie
-- 
2.35.2.windows.1

